### PR TITLE
v1.8 backports 2021-03-12

### PR DESCRIPTION
**Pull request by kaworu**
* #15263 -- node: Fix CIDR comparison when updating routes (@brb)
 * #15194 -- azure: populate ApplicationSecurityGroups field in the new IPConfigurations (@AnishShah)
 * #15303 -- ipam: Fix ENI routing for secondary CIDRs (@gandro)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 15263 15194 15303; do contrib/backporting/set-labels.py $pr done 1.8; done
```

### CODE REVIEW

**Comment by kaworu** [comment]
#15194 (azure: populate ApplicationSecurityGroups field in the new IPConfigurations) looks good, all patches applied cleanly.

**Comment by kaworu** [comment]
test-backport-1.8

**Review by brb — APPROVED** [review]
:+1: for my changes, thanks.

File: pkg/aws/ec2/ec2.go [hunk 1]
```diff
@@ -268,6 +269,12 @@ func (c *Client) GetVpcs(ctx context.Context) (ipamTypes.VirtualNetworkMap, erro
 			vpc.PrimaryCIDR = *v.CidrBlock
 		}
 
+		for _, c := range v.CidrBlockAssociationSet {
+			if cidr := aws.ToString(c.CidrBlock); cidr != vpc.PrimaryCIDR {
```
Comment (gandro) [inline pkg/aws/ec2/ec2.go]: This I believe won't compile on 1.8, it needs to be
```suggestion
			if cidr := *c.CidrBlock; cidr != vpc.PrimaryCIDR {
```

Edit: Or `aws.StringValue`

Comment (kaworu) [reply pkg/aws/ec2/ec2.go]: I used `aws.StringValue` just the same as for [the v1.9 backport](https://github.com/cilium/cilium/pull/15331#issuecomment-799249798), should be ok now.

**Comment by kaworu** [comment]
test-backport-1.8

**Review by jrajahalme — APPROVED** [review]

**Review by gandro — APPROVED** [review]

**Comment by kaworu** [comment]
test-backport-1.8

**Comment by kaworu** [comment]
test-1.18-4.9

**Comment by kaworu** [comment]
test-runtime

**Comment by kaworu** [comment]
Adding the `ready-to-merge` as all reviewer have approved the PR and the CI passed.

**Pull request merged**